### ISSUE TITLE
define-by-run style for gotgonia?

### ISSUE THREAD

**Issue by LumRamabaja** [gorgonia/gorgonia#449]
As described in the documentation, Gorgonia seems to use a "define-then-run" style of computation (like in TF). Is there a way to use Gorgonia in a "define-by-run" style, like e.g. in Pytorch? To elaborate my question a bit:
In my specific case, I will have to use loops and if statements during forward propagation, i.e. each propagation might result in a different "path" in my model. With the define-by-run style of Pytorch, it seems pretty straight forward how to implement that, but I'm not sure if using loops and if statements is even possible with the static graph approach. 

So in a nutshell: Is it possible to dynamically build a graph at runtime like in Pytorch?

**Comment by LumRamabaja** [comment]
Nevermind, found it https://gorgonia.org/reference/vm/lispmachine/

**Comment by chewxy** [comment]
And in the next version, it'll be even better: 

https://github.com/gorgonia/gorgonia/blob/v0.10.0-working/exprgraph/examples_test.go#L193-L321
